care/facility/api/serializers/patient_consultation.py

```diff
@@ -149,12 +151,14 @@ def update(self, instance, validated_data):
 
             if "review_interval" in validated_data:
                 review_interval = validated_data.pop("review_interval")
+                instance.review_interval = review_interval
                 if review_interval >= 0:
-                    instance.review_interval = review_interval
                     instance.save()
                     patient.review_time = localtime(now()) + timedelta(
                         minutes=review_interval
                     )
+                else:
+                    patient.review_time = None
```

Comment:
for `else` block the instance is not saved
bring `instance.save()` outside conditional block